## End-to-end

This notebook shows an end-to-end `aiice`

In [16]:
import torch
from datetime import datetime, timedelta
import matplotlib.pyplot as plt

In [2]:
from aiice.loader import Loader
from aiice.preprocess import SlidingWindowDataset, apply_downsample

## Data Preparation

Let's get data for past 5 years and split them:
- 2025 year - test data
- 2024 year - val data
- 2020-2023 - train data

Also, we're going to get date per 7 days in order to simplify our example.
Our model will predict next 4 days by last 12 days per 7 days (next 4 weeks by last 12 weeks)

In [10]:
start = datetime(2020, 1, 1)
step = 7
pre_history_len = 12
forecast_len = 4

In [5]:
step = 7
start = datetime(2020, 1, 1)

loader = Loader()
data = loader.get(start=start.date(), step=step, tensor_out=True)

global_series/2020/osisaf_20200122.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200226.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200212.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200115.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200415.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200401.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200304.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200101.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200219.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200129.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200311.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200108.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200318.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200422.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200408.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200325.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200205.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200429.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200610.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200624.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200603.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200617.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200527.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200506.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200715.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200722.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200701.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200805.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200812.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200513.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200520.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200729.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200826.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200902.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200916.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200708.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200930.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201021.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200909.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201028.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201125.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200819.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201007.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201104.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20200923.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201111.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210120.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201202.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201216.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201118.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201230.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210217.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210113.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201209.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201223.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210303.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2020/osisaf_20201014.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210310.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210106.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210210.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210127.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210324.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210224.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210203.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210414.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210421.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210505.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210331.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210512.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210407.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210317.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210428.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210609.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210623.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210519.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210630.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210707.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210616.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210526.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210811.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210714.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210908.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210922.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210721.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210901.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210818.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210929.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210915.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210825.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210602.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211124.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211013.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210804.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20210728.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211117.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211208.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211006.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211103.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211222.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211027.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211215.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211229.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220119.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220126.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211020.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220112.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211110.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2021/osisaf_20211201.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220223.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220202.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220316.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220406.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220309.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220330.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220209.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220216.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220105.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220504.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220302.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220511.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220323.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220413.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220427.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220608.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220518.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220629.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220706.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220420.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220803.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220601.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220720.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220615.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220727.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220824.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220817.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220622.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220907.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221005.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221019.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220810.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220525.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220831.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221012.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221102.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220713.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221116.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220914.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221123.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221026.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221221.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230104.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221109.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221207.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230111.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221228.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220928.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221130.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230125.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230201.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20220921.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2022/osisaf_20221214.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230118.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230222.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230308.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230329.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230215.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230405.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230412.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230322.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230510.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230503.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230524.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230301.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230208.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230419.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230315.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230802.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230726.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230531.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230426.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230517.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230614.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230628.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230712.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230705.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230809.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230816.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230621.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230913.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230607.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230830.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231025.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230719.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231101.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231004.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230927.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231011.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231018.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230823.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230906.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20230920.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231220.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231129.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231227.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231206.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240131.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240214.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231213.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231115.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240117.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240110.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240228.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240207.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240306.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240124.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240327.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240103.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231122.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240417.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2023/osisaf_20231108.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240403.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240320.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240522.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240515.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240221.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240508.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240612.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240424.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240710.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240605.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240619.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240410.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240703.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240724.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240731.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240529.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240814.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240501.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240918.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241002.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240904.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240313.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241016.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241023.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241113.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240828.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241106.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241030.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241120.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241009.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241127.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240925.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241211.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240821.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240626.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241218.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241225.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250108.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250129.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240911.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240717.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250122.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20240807.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250205.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250212.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250226.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2024/osisaf_20241204.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250305.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250219.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250101.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250319.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250402.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250115.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250326.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250528.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250416.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250312.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250507.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250430.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250618.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250521.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250604.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250625.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250409.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250423.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250514.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

global_series/2025/osisaf_20250611.npy:   0%|          | 0.00/747k [00:00<?, ?B/s]

In [6]:
data.shape

torch.Size([287, 432, 432])

Also let's reduce our tensors with `apply_downsample` in order to minimize compute time of out future model

In [8]:
# delete each 2nd pixel by 1 and 2 axes
downsampled_date = apply_downsample(data, i=2, axes=(1, 2))
downsampled_date.shape

torch.Size([287, 216, 216])

For now we have 287 days, with the code below we can split them:

In [11]:
train_idx, val_idx, test_idx = [], [], []

for i in range(len(downsampled_date)):
    current_date = start + timedelta(days=i*step)
    year = current_date.year
    
    if 2020 <= year <= 2023:
        train_idx.append(i)
    elif year == 2024:
        val_idx.append(i)
    elif year == 2025:
        test_idx.append(i)

train = downsampled_date[train_idx]
val = downsampled_date[val_idx]
test = downsampled_date[test_idx]

print(train.shape, val.shape, test.shape)

torch.Size([209, 216, 216]) torch.Size([52, 216, 216]) torch.Size([26, 216, 216])


In [ ]:
x, y = dataset[0]

plt.figure()
plt.title("X[0]")
plt.imshow(x[0])

plt.figure()
plt.title("Y[-1]")
plt.imshow(y[-1])
plt.show()

The last step is to form datasets:

In [12]:
train_dataset = SlidingWindowDataset(
    data=train,
    pre_history_len=pre_history_len,
    forecast_len=forecast_len,
)

val_dataset = SlidingWindowDataset(
    data=val,
    pre_history_len=pre_history_len,
    forecast_len=forecast_len,
)

test_dataset = SlidingWindowDataset(
    data=test,
    pre_history_len=pre_history_len,
    forecast_len=forecast_len,
)

In [13]:
print(len(train_dataset), train_dataset[0][0].shape, train_dataset[0][1].shape)

194 torch.Size([12, 216, 216]) torch.Size([4, 216, 216])


In [14]:
print(len(val_dataset), val_dataset[0][0].shape, val_dataset[0][1].shape)

37 torch.Size([12, 216, 216]) torch.Size([4, 216, 216])


In [15]:
print(len(test_dataset), test_dataset[0][0].shape, test_dataset[0][1].shape)

11 torch.Size([12, 216, 216]) torch.Size([4, 216, 216])


## Train model

## Model benchmark